pkg/kp/lock.go

```diff
@@ -86,6 +117,23 @@ func (l Lock) Lock(key string) error {
 	return AlreadyLockedError{Key: key}
 }
 
+func (l Lock) continuallyRenew() {
+	for {
+		select {
+		case <-l.renewalCh:
+			err := l.Renew()
+			if err != nil {
+				l.renewalErrCh <- err
+				l.Destroy()
+				return
+			}
+		case <-l.quitCh:
+			close(l.renewalErrCh)
```

Comment:
Could you boost this to the top of the method as `defer close(l.renewalErrCh)`?  That give the `Lock` object a nice invariant: the error channel will always be closed once its producer quits producing.